src/consensus/zerocoin_verify.cpp

```diff
@@ -0,0 +1,460 @@
+// Copyright (c) 2020 The PIVX developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "zerocoin_verify.h"
+
+#include "chainparams.h"
+#include "consensus/consensus.h"
+#include "guiinterface.h"        // for ui_interface
+#include "init.h"                // for ShutdownRequested()
+#include "invalid.h"
+#include "main.h"
+#include "script/interpreter.h"
+#include "spork.h"               // for sporkManager
+#include "txdb.h"
+#include "utilmoneystr.h"        // for FormatMoney
+
+bool CheckZerocoinSpend(const CTransaction& tx, bool fVerifySignature, CValidationState& state, bool fFakeSerialAttack)
+{
+    //max needed non-mint outputs should be 2 - one for redemption address and a possible 2nd for change
+    if (tx.vout.size() > 2) {
+        int outs = 0;
+        for (const CTxOut& out : tx.vout) {
+            if (out.IsZerocoinMint())
+                continue;
+            outs++;
+        }
+        if (outs > 2 && !tx.IsCoinStake())
+            return state.DoS(100, error("CheckZerocoinSpend(): over two non-mint outputs in a zerocoinspend transaction"));
+    }
+
+    //compute the txout hash that is used for the zerocoinspend signatures
+    CMutableTransaction txTemp;
+    for (const CTxOut& out : tx.vout) {
+        txTemp.vout.push_back(out);
+    }
+    uint256 hashTxOut = txTemp.GetHash();
+
+    bool fValidated = false;
+    std::set<CBigNum> serials;
+    int64_t nTotalRedeemed = 0;
```

Comment:
This object was moved to int64_t when was a CAmount.

Reply:
fixed